main/Main.hs

```diff
@@ -40,8 +43,12 @@ formatStdio w = do
 
 formatFile :: Int -> FilePath -> IO Result
 formatFile w path = do
-    contents <- TextIO.readFile path
-    mapM (atomicWriteFile path) $ format w path contents
+    contents <- readFileUtf8 path
+    mapM atomicWriteFile $ format w path contents
+  where
+    atomicWriteFile t = withOutputFile path $ \h -> do
+      hSetEncoding h utf8
```

Comment:
Is it possible to make the two new modules more composable so no implementation details leak here?

Reply:
I don’t think so, and I also don’t think any implementation details leak here. Correct handling of the data encoding is ultimately the responsibility of the programmer, the utf8 module just has helpers for frequently used tasks that can be considered “best practices”.

Reply:
Okay, it can't be helped then.